Update enumset dependency to enable defmt

chrysn opened the pull request:
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
  **Not applicable**: This enables new functionality that would not be covered by examples yet (nor does it need to).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
  **Not applicable**: No code was changed.
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
  **Unsure if needed**: A minor bump in a dependency would probably not be worth mentioning, and neither is the feature forwarding (we wouldn't mention other new features of a dependency either, this just needs a tad more work to become usable).
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
  **Not applicable**: No changes needed downstream.
- [x] My changes are in accordance to the [esp-rs developer guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/DEVELOPER-GUIDELINES.md)
   **Does not really apply, that is about APIs and code and not about **


#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

  Note that editing is sadly not possible due to GitHub's limitations; I'll try to be quick about applying any suggestions.

### Pull Request Details 📖

#### Description

`enumset::EnumSet` is part of this crate's [public API](https://docs.esp-rs.org/esp-hal/esp-wifi/0.12.0/esp32/esp_wifi/wifi/struct.WifiController.html#method.capabilities), but did not implement defmt even though this crate has a flag for it.

Updating enumset by a minor revision enables cross-linking the defmt features to gain an implementation of `Format` on the capabilities.

#### Testing

I think that CI passing should suffice (but don't know this project's CI coverage to be sure).

bjoernQ commented:
Not important but just for consistency we should also bump the patch-version of enumset in esp-hal

jessebraham commented:
`esp-alloc` also depends on this package.

MabezDev reviewed (changes requested):
Sorry for the delay, we were at RustWeek :D.

Could you please update the all the places we use enumset please? After that we can merge this.

chrysn commented:
Sorry, I don't understand: What kind of updates would be needed there? The types are still named the same, it's a newer crate, and that the type now (conditionally) has one more impl on it is not written (and thus changeable) in any piece of the code that I'm aware of.

MabezDev reviewed (approved):
Thank you!

chrysn commented:
I've updated the other crates too. Both other crates have EnumSet types as part of their public interface; consequently, defmt gets forwarded as enumset/defmt there as well.

The contributing guide doesn't state: Do you want the commits squashed right away? (So far I've added the change as a fixup commit in the PR; really those should be one commit.)

MabezDev commented:
> I've updated the other crates too. Both other crates have EnumSet types as part of their public interface; consequently, defmt gets forwarded as enumset/defmt there as well.
> 
> The contributing guide doesn't state: Do you want the commits squashed right away? (So far I've added the change as a fixup commit in the PR; really those should be one commit.)

Our merge strategy for the queue is the squash, so don't worry about the git history being perfect :).
